### PR TITLE
Completed Your Order Please challenge

### PR DESCRIPTION
**Pull request by nxnode**
look at this solution that was top of the list:

` return ' '.join(sorted(words.split(), key=lambda w:sorted(w)))`

### CODE REVIEW

File: code_challenges/code_wars/your_order_please.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# https://www.codewars.com/kata/55c45be3b2079eccff00010f
+
+
+def sort_order(order):
+    numbers = {1, 2, 3, 4, 5, 6, 7, 8, 9}
```
Comment (W-Ely) [inline code_challenges/code_wars/your_order_please.py:5]: No blocking observation - no need to change. These are only used as strings, so could be strings here like `numbers = {"1", "2", "3", "4", "5", "6", "7", "8", "9"}`

**Review by W-Ely — APPROVED** [review]
`return ' '.join(sorted(words.split(), key=lambda w:sorted(w)))`
^ is a clever use of sorted putting numbers before letters in the key function.

**Comment by nxnode** [comment]
> 

I don't understand how this is actually doing what it is doing. What exactly is filtering for numbers to get them in front?

**Comment by W-Ely** [comment]
> I don't understand how this is actually doing what it is doing. What exactly is filtering for numbers to get them in front?

This goes back to what the sorted functions `key` keyword argument takes.  It takes a function which returns something that should be used to compare or sort by.
```python
l = ["a2b", "b1c"]
n = sorted(l)
print("n", n)
n = sorted(l, key=lambda x: sorted(x))
print(n)
```

These 2 functions are equivilant. (But the first namless function stored in a variable isn't something you should do....)
```python
sort_func_lambda = lambda x: sorted(x)

def sort_func_def(x):
    return sorted(x)

l = ["a2b", "b1c"]
n = sorted(l, key=sort_func_lambda)
print("n", n)
n = sorted(l, key=sort_func_def)
print("n", n)
```

So what is the lambda or the def actually returning?

```python
def sort_func_def(x):
    sorted_x = sorted(x)
    print("sorted_x", sorted_x)
    return sorted_x

l = ["a2b", "b1c"]
n = sorted(l, key=sort_func_def)
print("n", n)
```

In this case, its returning lists with the letters from each entry sorted, so with numbers before letters. And apparently Python has an idea about how to compare lists to see which is greater.
```python
print(["2", "a", "b"] < ["1", "b", "c"])
```

**Pull request merged**